crates/red_knot/src/symbols.rs

```diff
@@ -439,7 +489,7 @@ impl SymbolTableBuilder {
                     ast::TypeParam::ParamSpec(ast::TypeParamParamSpec { name, .. }) => name,
                     ast::TypeParam::TypeVarTuple(ast::TypeParamTypeVarTuple { name, .. }) => name,
                 };
-                self.add_symbol(name);
+                self.add_or_update_symbol(name, SymbolFlags::IS_USED);
```

Comment:
Oh, this isn't something we ever discussed, but type params are defined by the annotation scope that introduces them. And we should test for this case as well. (There are already tests for generic func and generic class.)
```suggestion
                self.add_or_update_symbol(name, SymbolFlags::IS_DEFINED);
```

Reply:
Great; was wondering about that! I'll add a test.